packages/parameterized_test/lib/src/test_parameters.dart

```diff
@@ -0,0 +1,323 @@
+//ignore_for_file: camel_case_types
+
+import 'util/cast_helpers.dart';
+
+/// alias for [TestParameters1].
+typedef p1<A1> = TestParameters1<A1>;
+
+/// alias for [TestParameters2].
+typedef p2<A1, A2> = TestParameters2<A1, A2>;
+
+/// alias for [TestParameters3].
+typedef p3<A1, A2, A3> = TestParameters3<A1, A2, A3>;
+
+/// alias for [TestParameters4].
+typedef p4<A1, A2, A3, A4> = TestParameters4<A1, A2, A3, A4>;
+
+/// alias for [TestParameters5].
+typedef p5<A1, A2, A3, A4, A5> = TestParameters5<A1, A2, A3, A4, A5>;
+
+/// alias for [TestParameters6].
+typedef p6<A1, A2, A3, A4, A5, A6> = TestParameters6<A1, A2, A3, A4, A5, A6>;
+
+/// alias for [TestParameters7].
+typedef p7<A1, A2, A3, A4, A5, A6, A7>
+    = TestParameters7<A1, A2, A3, A4, A5, A6, A7>;
+
+/// alias for [TestParameters8].
+typedef p8<A1, A2, A3, A4, A5, A6, A7, A8>
+    = TestParameters8<A1, A2, A3, A4, A5, A6, A7, A8>;
+
+/// alias for [TestParameters9].
+typedef p9<A1, A2, A3, A4, A5, A6, A7, A8, A9>
+    = TestParameters9<A1, A2, A3, A4, A5, A6, A7, A8, A9>;
+
+/// alias for [TestParameters10].
+typedef p10<A1, A2, A3, A4, A5, A6, A7, A8, A9, A10>
+    = TestParameters10<A1, A2, A3, A4, A5, A6, A7, A8, A9, A10>;
+
+/// Provides a interface for implementing test body classes.
+abstract class TestParameters {
+  abstract final Function body;
+  abstract final int count;
+
+  void mapBody<R>(Iterable<R> values);
+}
+
+/// Test body class that handles 1 test parameter value.
+/// For example:
+/// ```dart
+/// TestParameter1((int number){
+///   // Do some test logic here.
+/// });
+/// ```
+class TestParameters1<A1> implements TestParameters {
+  const TestParameters1(this.body);
+
+  @override
+  final dynamic Function(A1) body;
+
+  @override
+  final int count = 1;
+
+  @override
+  void mapBody<R>(Iterable<R> values) {
```

Comment:
What do you think of adding a Assert to these bodies to check if the amount of values you receive is the same as the method expects?

Reply:
Ah I see you test this when mapping the tests!